api/features_api.py

```diff
@@ -29,9 +29,25 @@ class FeaturesAPI(basehandlers.APIHandler):
 
   def do_get(self):
     user = users.get_current_user()
-    feature_list = models.Feature.get_chronological(
-        version=2,
-        show_unlisted=permissions.can_edit_feature(user, None))
+    feature_list = None
+
+    # Query-string parameter 'milestone' is provided
+    if self.request.args.get('milestone') is not None:
+      try:
+        milestone = int(self.request.args.get('milestone'))
+        feature_list = models.Feature.get_in_milestone(
+          version=2,
+          show_unlisted=permissions.can_edit_feature(user, None), 
```

Comment:
The call to permissions.can_edit_feature(user, None) could be factored out to the top of the method.

Reply:
Done. Thank you for pointing this out.